app/lib/api/file_system/file_system_remote.dart

```diff
@@ -143,10 +143,30 @@ mixin RemoteSystem {
   }
 
   Future<void> clearCachedContent() async {
-    var cacheDir = await getRemoteCacheDirectory();
-    var directory = Directory(cacheDir);
-    if (await directory.exists()) {
-      await directory.delete(recursive: true);
+    try {
+      var cacheDir = await getRemoteCacheDirectory();
+      var directory = Directory(cacheDir);
+      final exists = await directory.exists();
+      if (exists) {
+        try {
+          await directory.delete(recursive: true);
+          // Directory deleted successfully, perform any needed actions
+        } on FileSystemException catch (e) {
+          if (e.osError?.errorCode == 32) {
+            // Directory in use, retry after a short delay
+            await Future.delayed(const Duration(seconds: 1));
+            await clearCachedContent();
+          } else if (e.osError?.errorCode == 2) {
+            // Directory not found, no further action needed
+          } else {
+            // Handle unexpected FileSystemException
+          }
+        }
+      } else {
+        // No directory exists, no action needed
+      }
+    } catch (e) {
+      // Handle any other exceptions that may occur
```

Comment:
I wouldn't put it in a try catch, the method that called this function should know that there is an exception happened